pkg/topology/kademlia/kademlia.go

```diff
@@ -844,11 +844,6 @@ func binSaturated(oversaturationAmount int, staticNode staticPeerFunc, k *Kad) b
 // recalcDepth calculates, assigns the new depth, and returns if depth has changed
 func (k *Kad) recalcDepth() {
 
-	if k.storageRadius != swarm.MaxPO {
-		k.depth = k.storageRadius
```

Comment:
Previously, we made this change to force pullsync to sync at storage radius, not connection depth.

@metacertain pointed out that this is not possible because at a high storage radius, there may no peers in the neighborhood.

So this draft PR makes the protocols use batchstore directly (depending on if its's a full node or not).

Reply:
@aloknerurkar @metacertain  ^^^